tools/crm_mon.h

```diff
@@ -58,27 +58,26 @@ typedef enum mon_output_format_e {
     mon_output_cgi
 } mon_output_format_t;
 
-#define mon_show_times         (0x0001U)
-#define mon_show_stack         (0x0002U)
-#define mon_show_dc            (0x0004U)
-#define mon_show_count         (0x0008U)
-#define mon_show_nodes         (0x0010U)
-#define mon_show_resources     (0x0020U)
-#define mon_show_attributes    (0x0040U)
-#define mon_show_failcounts    (0x0080U)
-#define mon_show_operations    (0x0100U)
-#define mon_show_tickets       (0x0200U)
-#define mon_show_bans          (0x0400U)
-#define mon_show_fence_history (0x0800U)
-
-#define mon_show_headers       (mon_show_times | mon_show_stack | mon_show_dc \
-                               | mon_show_count)
-#define mon_show_default       (mon_show_headers | mon_show_nodes \
-                               | mon_show_resources)
-#define mon_show_all           (mon_show_default | mon_show_attributes \
-                               | mon_show_failcounts | mon_show_operations \
-                               | mon_show_tickets | mon_show_bans \
-                               | mon_show_fence_history)
+#define mon_show_stack          (1 << 0)
+#define mon_show_dc             (1 << 1)
+#define mon_show_times          (1 << 2)
+#define mon_show_counts         (1 << 3)
+#define mon_show_options        (1 << 4)
+#define mon_show_nodes          (1 << 5)
+#define mon_show_resources      (1 << 6)
+#define mon_show_attributes     (1 << 7)
+#define mon_show_failcounts     (1 << 8)
+#define mon_show_operations     (1 << 9)
+#define mon_show_fencing        (1 << 10)
+#define mon_show_tickets        (1 << 11)
+#define mon_show_bans           (1 << 12)
+
+#define mon_show_summary        (mon_show_stack | mon_show_dc | mon_show_times | \
+                                 mon_show_counts | mon_show_options)
+#define mon_show_default        (mon_show_summary | mon_show_nodes | mon_show_resources)
```

Comment:
The default will have to vary by format to preserve the behavior: mon_show_options should be in html defaults but not text/console. Since we don't know the format until after parsing the options, I'm thinking maybe include/exclude should just be string options rather than callbacks, and after parsing, set mon_show_default per format, then do the equivalent of the callbacks.

That would also make it easy to enable --exclude for xml format.

Reply:
I think I've got this figured out, but this is going to require an awful lot of testing before I am confident in it.

Reply:
Confirmed that --exclude works for XML, and that you get everything if you don't specify any --include or --exclude option.